connectors/sources/onedrive.py

```diff
@@ -253,44 +270,90 @@ async def close_session(self):
         strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
         skipped_exceptions=NotFound,
     )
-    async def get(self, url):
+    async def get(self, url, header=None):
         access_token = await self.token.get()
         headers = {"authorization": f"Bearer {access_token}"}
+        if header:
+            headers |= header
         try:
             async with self.session.get(url=url, headers=headers) as response:
                 yield response
         except ServerConnectionError:
             await self.close_session()
             raise
         except ClientResponseError as e:
-            if e.status == 429 or e.status == 503:
-                response_headers = e.headers or {}
-                retry_seconds = DEFAULT_RETRY_SECONDS
-                if "Retry-After" in response_headers:
-                    try:
-                        retry_seconds = int(response_headers["Retry-After"])
-                    except (TypeError, ValueError) as exception:
-                        self._logger.error(
-                            f"Error while reading value of retry-after header {exception}. Using default retry time: {DEFAULT_RETRY_SECONDS} seconds"
-                        )
-                else:
-                    self._logger.warning(
-                        f"Rate Limited but Retry-After header is not found, using default retry time: {DEFAULT_RETRY_SECONDS} seconds"
+            await self._handle_client_side_errors(e)
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+        skipped_exceptions=NotFound,
+    )
+    async def post(self, url, payload=None):
+        access_token = await self.token.get()
+        headers = {
+            "authorization": f"Bearer {access_token}",
+            "Content-Type": "application/json",
+        }
+        try:
+            async with self.session.post(url, headers=headers, data=payload) as resp:
+                if url.endswith("/$batch"):
+                    await self._check_batch_items_throttled(resp)
+                yield resp
+
+        except ClientResponseError as e:
+            await self._handle_client_side_errors(e)
+        except ClientPayloadError as e:
```

Comment:
The name is a bit misleading 😄 . From the official [docs](https://docs.aiohttp.org/en/stable/client_reference.html): 

> Reading from the stream may raise [aiohttp.ClientPayloadError](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientPayloadError) if the response object is closed before response receives all data or in case if any transfer encoding related errors like malformed chunked encoding of broken compression data.